fix: undefined default search bug

Falcasantos opened the pull request:
checks to see if `search` is undefined; if so, then returns a welcome message

Falcasantos committed:
fix: undefined default search bug

checks to see if `search` is undefined; if so, then returns a welcome message

juliuscc commented on container/Exchange.js:
```diff
@@ -49,11 +53,19 @@ export default () => {
-        <Translation translationResult={translation} />
-      )}
+      {(() => {
+        let element = null
```
Wouldn't it make more sense to just return in the if/else-cases?

Falcasantos replied:
nevermind yes!

juliuscc commented on container/Exchange.js:
```diff
@@ -49,11 +53,19 @@ export default () => {
-      )}
+      {(() => {
+        let element = null
+        if (search === undefined) {
```
This is the same as (!loading && !translation). We could bring back the if-else 4-way path that we had before

Falcasantos replied:
Actually when you load the website with no search queries, `loading` is defaulted to `true`, hence why I had to check `search` instead; `!loading && !translation` would never equal true in the default search case of no search params

juliuscc replied:
Maybe loading should default to false. That would make more sense, and only set loading to true if we actually do a search

Falcasantos replied:
Did so in the latest commit but I still put in a `|| search === ''`, so really is there any gain in doing `(!loading && !translation) || search === ''` over `!search`?

juliuscc replied:
I agree

juliuscc replied:
In a later PR we could refactor this into a state enum like this https://kentcdodds.com/blog/stop-using-isloading-booleans

We could also add error handling

juliuscc commented on container/Exchange.js:
```diff
@@ -32,6 +32,10 @@ export default () => {
   const [translation, setTranslation] = useState(undefined)
```
I think it would be good to use null here as well.

Falcasantos replied:
Donederydoo

juliuscc commented on container/Exchange.js:
```diff
@@ -32,6 +32,10 @@ export default () => {
   const [translation, setTranslation] = useState(undefined)
 
   useEffect(() => {
+    if (search === undefined) {
```
Does this also work for `localhost:3000/?search=` or would `search` be an empty string in that case? What do you think of doing this instead: 
```suggestion
    if (!search) {
```

I think we catch more cases that way.

Falcasantos replied:
Yes good point!

juliuscc replied:
This looks like it was resolved by mistake

Falcasantos committed:
fix: simplifies to two checks for (!search) and (loading||!translation)

also:
-returns within if/else
-`setTranslation(undefined)` => `setTranslation(null)`
-`search===undefined` => `!search`

Falcasantos committed:
fix: simplifies checks to check for loading/translation/empty search

Commit:
Updates to look for !search instead of undefined

Co-Authored-By: Julius Recep Colliander Celik <julius.cc@hotmail.com>
